### PR TITLE
Add msvc test project

### PR DESCRIPTION
**Pull request by Brett208**
There should still be about 19 warnings associated with compiling GoogleTest. 

I added some more unit tests to get a feel for actually using Google Test.

### CODE REVIEW

**Comment by DanRStevens** [comment]
Alright, I made a few updates. Curiously, GitHub orders PR items by date. I committed a few local changes while reviewing. Since those dates are before when I published my review comments, they get listed before on the PR thread, even though I pushed them after. Thought I should mention it in case you miss seeing them.

I left the `int`/`char` thing for you. The test suite name implies `int`, but I think `char` would be more appropriate. I leave it to you to decide what to do.

File: test/Streams/MemoryStreamReader.test.cpp [hunk 1]
```diff
@@ -39,3 +39,40 @@ TEST_F(EmptyMemoryStreamReader, EmptySize) {
...
   EXPECT_EQ(0, stream.ReadPartial(nullptr, 0));
 }
+
+// Setup fixture
+class IntMemoryStreamReader : public ::testing::Test {
+public:
+	IntMemoryStreamReader() : stream(buffer.data(), buffer.size()) {}
+protected:
+	Stream::MemoryReader stream;
+	const std::array<int, 5> buffer{ 0, 1, 2, 3, 4 };
```
[the remark this answers is not in the document]
Comment (Brett208) [reply test/Streams/MemoryStreamReader.test.cpp]: MSVC doesn't provide the said warning. Thanks for cleaning it up. 

I was going to use char, however I couldn't figure out how to initialize a std::array with say all 'a' values or '1's for that matter, which was really frustrating. I'll fix it though.

File: test/Streams/MemoryStreamReader.test.cpp [hunk 2]
```diff
@@ -39,3 +39,40 @@ TEST_F(EmptyMemoryStreamReader, EmptySize) {
...
+};
+
+TEST_F(IntMemoryStreamReader, SeekOutOfBounds) {
+	EXPECT_THROW(stream.Seek(6), std::runtime_error);
+}
+
+TEST_F(IntMemoryStreamReader, ReadOutOfBounds) {
+	std::array<int, 6> destinationBuffer;
+	EXPECT_THROW(stream.Read(destinationBuffer), std::runtime_error);
```
[the remark this answers is not in the document]
Comment (Brett208) [reply test/Streams/MemoryStreamReader.test.cpp]: Hmm, good point! Making unit tests pass by mistake...

File: test/Streams/MemoryStreamReader.test.cpp [hunk 3]
```diff
@@ -39,3 +39,40 @@ TEST_F(EmptyMemoryStreamReader, EmptySize) {
...
+}
+
+TEST_F(IntMemoryStreamReader, SeekOutOfBoundsRelative) {
+	// Seeking out of bounds should return a Stream::Seekable to previous condition
+	EXPECT_THROW(stream.SeekRelative(6), std::runtime_error);
+	EXPECT_THROW(stream.SeekRelative(-1), std::runtime_error);
```
[the remark this answers is not in the document]
Comment (Brett208) [reply test/Streams/MemoryStreamReader.test.cpp]: Thanks for the ideas. Yeah, this test is actually supposed to test both that an exception is raised when seeking out of bounds and that the stream is returned in its original form when recovering from the exception.

I definitely implemented it half-baked both in documentation and actual test.

**Comment by Brett208** [comment]
Thanks for the quick and thorough review of the code. That comment should have read `// Check for strong exception safety`. However, it is not needed now with the updated and better test names.

OP2UtilityTest does reference OP2Utility. It is hard to spot when reading the XML with all the clutter in Visual Studio project files.

```
  <ItemGroup>
    <ProjectReference Include="..\OP2Utility.vcxproj">
      <Project>{980d53d9-f9e2-4682-9307-1303c9e42313}</Project>
    </ProjectReference>
  </ItemGroup>
```

Thanks for fixing the spaces/tab issues. I had seen that and then forgot to correct it. 

Hopefully the changes I made are acceptable. If not, (or even if they are fine) we should open a dedicated branch for updating MemoryReader unit tests as I think this branch has run its course.

-Brett

**Comment by DanRStevens** [comment]
Ahh yes, "strong exception safety". That's a good description to include.

We should perhaps group together tests related to strong exception safety. Many streams may want to test that, though likely not every stream will want to guarantee it.

Good to have this merged in. Over the next little while, I'd like to focus on a number of small branches to add new tests to existing code.

**Pull request merged**